homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,211 @@
+
+        # Add Stelpro Specific
+        import bellows.types as t
+        # pylint: disable=W0212
+        self._endpoint.thermostat._attridx['setpoint_mode_manuf_specific'] = \
+            0x401C
+        # pylint: enable=W0212
+        self._endpoint.thermostat.attributes[0x401C] = \
+            ('setpoint_mode_manuf_specific', t.int8s)
+        self._endpoint.thermostat.attributes[0x001C] = \
+            ('system_mode', t.uint8_t)
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+
+        @asyncio.coroutine
+        def safe_read(cluster, attributes, raw=False):
+            """Swallow all exceptions from network read.
+            If we throw during initialization, setup fails. Rather have an
+            entity that exists, but is in a maybe wrong state, than no entity.
+            """
+            try:
+                result, _ = yield from cluster.read_attributes(
+                    attributes,
+                    allow_cache=False,
+                    raw=raw,
+                )
+                return result
+            except Exception:  # pylint: disable=broad-except
+                return {}
+
+        _LOGGER.debug("%s async_update", self.entity_id)
+
+        result = yield from safe_read(self._endpoint.thermostat,
+                                      ['local_temp',
+                                       'pi_heating_demand',
+                                       'occupied_heating_setpoint',
+                                       'setpoint_mode_manuf_specific'])
+        self._heating_power = int(result.get('pi_heating_demand'))
+        raw_mode = result.get('setpoint_mode_manuf_specific', {})
+        self._current_operation = self._modes.get(raw_mode)
+
+        raw_temp = result.get('occupied_heating_setpoint')
+        if raw_temp:
+            self._target_temperature = raw_temp / 100.0
+        raw_temp = result.get('local_temp')
+        if raw_temp:
+            self._current_temperature = raw_temp / 100.0
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if self._unit == 'C':
+            return TEMP_CELSIUS
+        elif self._unit == 'F':
+            return TEMP_FAHRENHEIT
+        return self._unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """Return a list of available operation modes."""
+        return self._operation_list
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self._target_temperature < 0:
+            return None
+        return self._target_temperature
+
+    @asyncio.coroutine
+    def async_set_temperature(self, **kwargs):
```

Comment:
According to https://home-assistant.io/components/climate/ , this call can also be used to set operation mode and temperature in one go. Your method seems to ignore the operation mode.

Reply:
Fixed